corehq/apps/sso/models.py

```diff
@@ -143,6 +143,13 @@ class IdentityProvider(models.Model):
     last_modified = models.DateTimeField(auto_now=True)
     last_modified_by = models.EmailField()
 
+    # for auto-deactivation web user purposes
+    enable_user_deactivation = models.BooleanField(default=False)
+    api_host = models.TextField(default="")  # tenant id
+    api_id = models.TextField(default="")  # application (client) id in Azure AD
+    api_secret = models.TextField(default="")
```

Comment:
Nit: How do we feel about `default=""` vs `blank=True`? I think they probably result in the same thing, but the semantics seem cleaner behind `blank=True`. If `enable_user_deactivation` is `False`, it doesn't feel like we should be supplying any value, not defaulting to `""`

Reply:
Hi Matt, [Daniel proposed we should avoid having null values in database, so we default to ""](https://docs.google.com/document/d/1LOF1lCl3wTsEQkEEsHtZi3u4N6kvjXaLv-Lu6e9aYjo/edit?disco=AAABEfth3Mo).
`default=""` is more like a database level attribute, while `blank=True` is more like a validation level attribute.
 `blank=True` is to [allow the field to be blank when the form is submitted](https://docs.djangoproject.com/en/5.0/ref/models/fields/#blank).Though blank is default to False, but without specifying `blank=True`, when `enable_user_deactivation` is `False`, and when other fields is left empty, no validation error is presented. So I think we don't need `blank=True`, we control the validation using [this function](https://github.com/dimagi/commcare-hq/blob/490c2278f6f8f23849f3239c9a603c7503fbf74a/corehq/apps/sso/forms.py#L64).

Reply:
I see, I think I misunderstood and thought that `blank` would also provide a default value. Given that we're modifying an existing model, I guess we need to either supply `NULL` or the empty string, and I agree that we'd want to avoid `NULL` here